panic in verifyOldVolumesInfo

Running with f83073d3eb23a3f3c22bef9401090df5da53cc2f, on startup `docker -d -D -s overlay` it panics with the below.

```
panic: assignment to entry in nil map

goroutine 31 [running]:
github.com/docker/docker/daemon.(*Daemon).verifyOldVolumesInfo(0xc20802aee0, 0xc2083c2d20, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/volumes.go:259 +0x6f8
github.com/docker/docker/daemon.(*Daemon).register(0xc20802aee0, 0xc2083c2d20, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/daemon.go:215 +0x339
github.com/docker/docker/daemon.func·017(0xc2083c2d20, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:330 +0x223
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 1 [semacquire]:
sync.(*WaitGroup).Wait(0xc20835f600)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/docker/docker/daemon.(*Daemon).restore(0xc20802aee0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/daemon.go:345 +0xaf1
github.com/docker/docker/daemon.NewDaemon(0x13efa00, 0xc208038918, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/daemon.go:886 +0x2a03
main.mainDaemon()
	/go/src/github.com/docker/docker/docker/daemon.go:143 +0x834
main.main()
	/go/src/github.com/docker/docker/docker/docker.go:92 +0x6b4

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 11 [chan receive]:
github.com/docker/docker/pkg/listenbuffer.(*defaultListener).Accept(0xc20808fb40, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/pkg/listenbuffer/buffer.go:73 +0xb6
net/http.(*Server).Serve(0xc2081070e0, 0x7f5a6f36dd50, 0xc20808fb40, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1728 +0x92
github.com/docker/docker/api/server.(*HttpServer).Serve(0xc20801d8e0, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/server/server.go:127 +0x4d
github.com/docker/docker/api/server.func·002(0xc20801d980, 0x4, 0xc20801d987, 0x14)
	/go/src/github.com/docker/docker/api/server/server.go:104 +0x1c1
created by github.com/docker/docker/api/server.(*Server).ServeApi
	/go/src/github.com/docker/docker/api/server/server.go:108 +0x512

goroutine 9 [chan receive]:
github.com/docker/docker/api/server.(*Server).ServeApi(0xc208034500, 0xc208096590, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/server/server.go:112 +0x590
main.func·003()
	/go/src/github.com/docker/docker/docker/daemon.go:129 +0x5f
created by main.mainDaemon
	/go/src/github.com/docker/docker/docker/daemon.go:135 +0x6bc

goroutine 10 [chan receive]:
github.com/docker/docker/daemon.func·021()
	/go/src/github.com/docker/docker/daemon/debugtrap.go:17 +0x5c
created by github.com/docker/docker/daemon.setupSigusr1Trap
	/go/src/github.com/docker/docker/daemon/debugtrap.go:20 +0x18e

goroutine 26 [chan receive]:
database/sql.(*DB).connectionOpener(0xc20816b0e0)
	/usr/local/go/src/database/sql/sql.go:589 +0x4c
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:452 +0x31c

goroutine 29 [chan receive]:
github.com/docker/docker/daemon.(*statsCollector).run(0xc2083ca5a0)
	/go/src/github.com/docker/docker/daemon/stats_collector.go:89 +0xb2
created by github.com/docker/docker/daemon.newStatsCollector
	/go/src/github.com/docker/docker/daemon/stats_collector.go:29 +0x116

goroutine 32 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c30e0, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 33 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c34a0, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 34 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c25a0, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 35 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c23c0, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 36 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c2780, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 37 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c2960, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 38 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c2b40, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 39 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c2f00, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 40 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c32c0, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 41 [runnable]:
github.com/docker/docker/daemon.func·017(0xc2083c21e0, 0xc208038b01)
	/go/src/github.com/docker/docker/daemon/daemon.go:319
created by github.com/docker/docker/daemon.(*Daemon).restore
	/go/src/github.com/docker/docker/daemon/daemon.go:343 +0xaba

goroutine 42 [runnable]:
runtime.timerproc()
	/usr/local/go/src/runtime/time.go:151
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1
```

--- a/daemon/daemon.go
+++ b/daemon/daemon.go
@@ -161,6 +161,7 @@ func (daemon *Daemon) load(id string) (*Container, error) {
 		CommonContainer: CommonContainer{
 			State:        NewState(),
 			root:         daemon.containerRoot(id),
+			MountPoints:  make(map[string]*mountPoint),
 			execCommands: newExecStore(),
 		},
 	}

